### PR TITLE
Escrow contract

### PR DESCRIPTION
**Pull request by fabriziogianni7**
# Description

creating escrow  using [minimal proxy pattern](https://www.rareskills.io/post/eip-1167-minimal-proxy-standard-with-initialization-clone-pattern)

the AI dev will call `cloneEscrow` on BBProxy contract  to clone an escrow contract.
this pattern allow to save a lot of money when deploying a new contract, as in our case we are creating an escrow for each new plan order. the escrow has a method `fundEscrow` for ai devs to send ERC20 tokens to the contract and a method `evadeOrder` called by a resolver account to release the escrow.

### Test BBEscrow.sol and BBProxy
1. Inside hardhat folder, create a .env (see .env.example)
2. run `npx hardhat test test/Proxy.ts`

### Deploy BBEscrow.sol and BBProxy
run `npx hardhat run ./scripts/deployBBContracts.ts --network <the network you want to deploy to>` (you need to have ETH to do this)

### CODE REVIEW

**Comment by fabriziogianni7** [comment]
still need to test fundEscrow, that's why is a draft.

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (juanmanso) [inline package-lock.json:1]: please remove this file whenever you have the chance

File: packages/hardhat/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,8876 @@
+{
```
Comment (juanmanso) [inline packages/hardhat/package-lock.json:1]: please use `yarn` as we are doing across the app (that's why there's a `yarn.lock` too) and remove this file

Comment (fabriziogianni7) [reply packages/hardhat/package-lock.json:1]: sure, will do. my bad

File: package-lock.json (hunk 1 above)
Comment (juanmanso) [reply package-lock.json:1]: Please don't forget about this one

**Review by juanmanso — APPROVED** [review]
LGTM, great job man! Left 2 small comments but all in all PR looks awesome

File: .gitignore [hunk 3]
```diff
@@ -131,4 +131,5 @@ dist
 .pnp.*
 
 # MacOs
-.DS_Store
\ No newline at end of file
+.DS_Store
+notes
```
[the remark this answers is not in the document]
Comment (fabriziogianni7) [reply .gitignore:135]: let's leave it this way

**Pull request merged**